Server

GovindSuresh opened the pull request:
Adds the cloud-config file

GovindSuresh committed:
Added the cloud-config file

This file will be used to configure new droplets made on Digital Ocean. 
Root SSH login has been removed to improve security and user accounts have been made for rkdr and GovindSuresh.
No password sudo access has been provided for now, but we may move to a different system at a later date

GovindSuresh committed:
Added comments describing what the config does

GovindSuresh committed:
basic corrections

GovindSuresh committed:
Correction to AllowUsers in the sshd-config file

GovindSuresh committed:
changed the sshd port back to 22 

Ease of use change for now. We can look to change the port to something else later.

GovindSuresh committed:
Added user-jon and public SSH key

rdkr commented on cloud-config.txt:
```diff
@@ -0,0 +1,61 @@
+
+#v1.0
+#Created two user accounts for the server admins. Automatically adds public keys for connection and provides passwordless
+#sudo access. The configuration also edits the sshd_config file to prevent root ssh access and allows ssh access from the two # users
```
can you update these docs and move them to the README, along with some usage instructions pls?

GovindSuresh replied:
Done friend

GovindSuresh committed:
minor changes

GovindSuresh committed:
Updated with description of the cloud-init file.

GovindSuresh committed:
updated comments

GovindSuresh committed:
Merge branch 'master' into server

GovindSuresh committed:
updated with runcmd commands to install docker

Does not currently work

GovindSuresh committed:
added the getdocker script installation method

GovindSuresh committed:
removed apt configurations for now

Used the script provided at get.docker.com to install via runcmd: 
Removed the apt configurations for now,

GovindSuresh committed:
fixed the source list for the docker repository

The config file now installs docker via the official docker repository using HTTPS .
It adds the users to the docker group to allow users to run docker without sudo.

GovindSuresh committed:
updated user ssh key

GovindSuresh committed:
minor fixes

GovindSuresh committed:
Merge branch 'master' into server

rdkr commented on README.md:
```diff
@@ -6,8 +6,20 @@ Project Code is a collaborative project to learn and teach Python, DevOps, and a
 ## Hosting
+### Cloud-init.txt 
+This is the cloud-init file that configures the server, it creates the main user accounts for the code-owners and provides passwordless sudo access via SSH. The file also edits the sshd_config file to prevent root ssh access.
+As we are using DigitalOcean, to use the cloud-init simply select the user-data box in the 'select additional options' section when creating a droplet and copy in the contents of th
```
wat?

rdkr commented on README.md:
```diff
@@ -6,8 +6,20 @@ Project Code is a collaborative project to learn and teach Python, DevOps, and a
+
+
+
+
```
pls trim to one blank line at end of file

rdkr commented on README.md:
```diff
@@ -6,8 +6,20 @@ Project Code is a collaborative project to learn and teach Python, DevOps, and a
+
 ## Hosting
+### Cloud-init.txt 
+This is the cloud-init file that configures the server, it creates the main user accounts for the code-owners and provides passwordless sudo access via SSH. The file also edits the sshd_config file to prevent root ssh access.
```
grammar

rdkr commented on cloud-config.txt:
```diff
@@ -0,0 +1,100 @@
+            keyserver: hkp://keyserver.ubuntu.com:80
+
+#installs packages from the repos defined above             
+runcmd:
```
`runcmd` is needed as well as `packages`?

rdkr reviewed (changes requested):
nice. pls check if `packages` and `runcmd` are trying to do same thing?

needs some tidying. also can we have a space between `#` and the comment for readability

GovindSuresh committed:
Removed run-cmd commands for installation

Tidied up comments

GovindSuresh committed:
Updated Readme with information on cloudinit

GovindSuresh committed:
minor changes to style

GovindSuresh committed:
minor style changes

GovindSuresh committed:
minor changes

GovindSuresh committed:
Removed power-state
